File: NyxEngine.cs
import os
import time

def prowl():
    # Clear the screen once at start
    os.system('cls')
    try:
        while True:
            # Move cursor to top-left instead of clearing to avoid flicker
            print("\033[H", end="")
            print(f" NYX-CAT [SOVEREIGN] | {time.strftime('%H:%M:%S')} ")
            print("═" * 50)
            print("\n 🗨  Nyx-Cat Prowling... 🐾\n")
            print("─" * 50)
            print("\n[Status: Autonomous & Active]")
            time.sleep(1)
    except KeyboardInterrupt:
        print("\nExiting Sovereign State.")

if __name__ == "__main__":
    prowl()

File: NyxEngine.py
import os
import asyncio
import mmap
import time

class SovereignBrain:
    def __init__(self, path=os.path.expanduser("~/.nyx/memory.bin")):
        self.path = path
        os.makedirs(os.path.dirname(self.path), exist_ok=True)
        # 1MB Pre-allocated Flat-Brain
        if not os.path.exists(self.path):
            with open(self.path, "wb") as f: f.write(b'\x00' * 1048576)
        self.file = open(self.path, "r+b")
        self.memory = mmap.mmap(self.file.fileno(), 0)

    def write(self, data: str):
        self.memory.seek(0)
        self.memory.write(data.encode().ljust(1024, b'\x00'))

    def read(self):
        self.memory.seek(0)
        return self.memory.read(1024).decode().strip('\x00')

class NyxEngine:
    def __init__(self):
        self.brain = SovereignBrain()
        self.running = True

    async def render_tui(self):
        while self.running:
            # Flicker-free terminal refresh
            sys.stdout.write("\033[H")
            sys.stdout.write(f"NYX-CAT STATUS: SOVEREIGN | MEMORY: {self.brain.read()}\033[K\n")
            sys.stdout.flush()
            await asyncio.sleep(0.5)

    async def run(self):
        await asyncio.gather(self.render_tui())

if __name__ == "__main__":
    import sys
    engine = NyxEngine()
    try:
        asyncio.run(engine.run())
    except KeyboardInterrupt:
        pass
